lib/srv/desktop/rdp/rdpclient/src/lib.rs

```diff
@@ -771,37 +837,118 @@ pub struct CGOSharedDirectoryAnnounce {
     pub name: *mut c_char,
 }
 
+#[derive(Debug)]
 pub struct SharedDirectoryAcknowledge {
-    pub err: u32,
+    pub err_code: u32,
     pub directory_id: u32,
 }
 
-/// CGOSharedDirectoryAcknowledge is a CGO-compatible version of
-/// the TDP Shared Directory Knowledge message that we pass back to Go.
 #[repr(C)]
 pub struct CGOSharedDirectoryAcknowledge {
-    pub err: u32,
+    pub err_code: u32,
     pub directory_id: u32,
 }
 
 impl From<SharedDirectoryAcknowledge> for CGOSharedDirectoryAcknowledge {
     fn from(ack: SharedDirectoryAcknowledge) -> CGOSharedDirectoryAcknowledge {
         CGOSharedDirectoryAcknowledge {
-            err: ack.err,
+            err_code: ack.err_code,
             directory_id: ack.directory_id,
         }
     }
 }
 
+#[derive(Debug)]
+pub struct SharedDirectoryInfoRequest {
+    completion_id: u32,
+    directory_id: u32,
+    path: String,
+}
+
+#[repr(C)]
+pub struct CGOSharedDirectoryInfoRequest {
+    pub completion_id: u32,
+    pub directory_id: u32,
+    pub path: *mut c_char,
+}
+
+impl From<ServerCreateDriveRequest> for SharedDirectoryInfoRequest {
+    fn from(req: ServerCreateDriveRequest) -> SharedDirectoryInfoRequest {
+        SharedDirectoryInfoRequest {
+            completion_id: req.device_io_request.completion_id,
+            directory_id: req.device_io_request.device_id,
+            path: req.path,
+        }
+    }
+}
+
+#[derive(Debug)]
+#[allow(dead_code)]
+pub struct SharedDirectoryInfoResponse {
+    completion_id: u32,
+    err_code: u32,
+    fso: FileSystemObject,
+}
+
+#[repr(C)]
+pub struct CGOSharedDirectoryInfoResponse {
```

Comment:
Why not put `repr(C)` on `SharedDirectoryInfoResponse`? 

Do we need to duplicate each struct?

Reply:
Yeah it seems like we maintain two variants, one CGO and one not CGO. TBH if we don't store them across calls this doesn't make a lot of sense, but having a duplicated struct with Rust owned data is needed if we want to keep it around.

Reply:
Technically no, however I think it will pay dividends to be explicit and disciplined about what data is for passing between Rust and Go (where we need to be quite diligent about memory) vs what data is for passing around like normal (where the runtime/compiler mostly prevents us from doing anything stupid).